providers/exoscale/exoscaleProvider.go

```diff
@@ -269,8 +269,8 @@ func (c *exoscaleProvider) createRecordFunc(rc *models.RecordConfig, domainID st
 			Priority: prio,
 		}
 
-		if rc.TTL != 0 {
-			ttl := int64(rc.TTL)
+		if rc.TTL.Value() != 0 {
+			ttl := int64(rc.TTL.Value())
```

Comment:
This is an example of where a provider was explicitly checking for 0, in a place where it could never be reached (as `dnscontrol` always replaced `0` with `DefaultTTL` before reaching this point). I presumed that this check was intentional and thus left it in place (as it is now triggerable).